base/test/valveModule_tests.cpp

```diff
@@ -0,0 +1,330 @@
+#include "stdafx.h"
+#include <boost/test/unit_test.hpp>
+#include <boost/filesystem.hpp>
+#include "ExternalSourceModule.h"
+#include "ExternalSinkModule.h"
+#include "ValveModule.h"
+#include "FrameMetadata.h"
+#include "Frame.h"
+#include "FrameContainerQueue.h"
+#include "Logger.h"
+#include "AIPExceptions.h"
+#include "test_utils.h"
+#include "Module.h"
+
+BOOST_AUTO_TEST_SUITE(valveModule_tests)
+
+class SinkModuleProps : public ModuleProps
+{
+public: 
+    SinkModuleProps() : ModuleProps()
+    {};
+};
+
+class SinkModule : public Module
+{
+public:
+    SinkModule(SinkModuleProps props) : Module(SINK, "sinkModule", props)
+    {};
+    boost::shared_ptr<FrameContainerQueue> getQue() { return Module::getQue(); }
+
+protected:
+    bool process() {};
+    bool validateOutputPins()
+    {
+        return true;
+    }
+    bool validateInputPins()
+    {
+        return true;
+    }
+};
+
+BOOST_AUTO_TEST_CASE(basic)
+{
+    auto source = boost::shared_ptr<ExternalSourceModule>(new ExternalSourceModule());
+    auto metadata = framemetadata_sp(new FrameMetadata(FrameMetadata::FrameType::GENERAL));
+    auto pinId = source->addOutputPin(metadata);
+    auto valve = boost::shared_ptr<ValveModule>(new ValveModule(ValveModuleProps(2)));
+    source->setNext(valve, true, false);
```

Comment:
Disabling sieve for source module has no meaning. For this connection, just regular setNext should also work

Reply:
The changes are made sieve disabled is removed for source, source module works with regular setNext.